exporters/prometheus/exporter.go

```diff
@@ -344,3 +356,116 @@ func sanitizeName(n string) string {
 
 	return b.String()
 }
+
+func (c *collector) validateMetrics(m metricdata.Metrics) (bool, metricdata.Metrics, error) {
```

Comment:
This feels like it duplicates a lot of the existing translation logic.  Instead, could we perform the validation after conversion?

Reply:
> Instead, could we perform the validation after conversion?

Does "perform the validation after conversion" mean that we call the validation function is each `addXXXMetric` function? Just like following:

```
func addHistogramMetric(ch chan<- prometheus.Metric, histogram metricdata.Histogram, m metricdata.Metrics ....) {
        // newly added validation 
        drop, m, err := validateMetric(m)
        if err != nil { otel.Handle(err) }
        if drop { return }

        for _, dp := range histogram.DataPoints {
        ...
        } 
}

func addSumMetric(ch chan<- prometheus.Metric, histogram metricdata.Histogram, m metricdata.Metrics ....) {
      	valueType := prometheus.CounterValue
	if !sum.IsMonotonic {
		valueType = prometheus.GaugeValue
	}
	if sum.IsMonotonic {
		// Add _total suffix for counters
		name += counterSuffix
	}

        // newly added validation 
        drop, m, err := validateMetric(m)
        if err != nil { otel.Handle(err) }
        if drop { return }

        for _, dp := range sum.DataPoints {
        ...
        } 
} 

```

@dashpole